rfcs/testing.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Testing standards
+
+---
+
+_@andrefcdias_
+
+## Summary
+
+The purpose of this RFC is to standardize our testing process and clarify the purpose of each of the testing layers we have. It is supposed to define what is the intended result from tests and why they are required.
+
+## Background
+
+This RFC came from the discussion of unit testing styles in components such as Flex and Text and the debate of having a valuable test vs a "correct" one.
+
+## Problem statement
+
+At the moment, everyone has their own testing standards and there is no common approach to problems like components that only apply styles and need to be tested.
+There is no majority decision on the solution for this problem, being split between Visual Regression tests (VR tests) and Unit Tests (UT).
+
+By defining a common approach, we can make the testing process less ambiguous, as well as provide a guide for the community to follow.
+
+## Detailed Design or Proposal
+
+To start, this is the proposal for defining the goals of each testing layer.
+
+- **Conformance**
+
+  Ensuring the consistency of our components. They test that props like `as`, `ref`, `className` are supported and that the component is rendered.
+
+  These can be achieved simply through the `isConformant` function generated when creating a converged component.
+
+- **DOM Snapshot**
+
+  Stricly DOM structure consistency enforcing. This will guarantee we do not introduce breaking changes for users who expect a certain structure to be rendered by our components.
```

Comment:
When do we want to test the DOM structure?
Are we planning to make the DOM structure a part of the API contract and consider any change to DOM a breaking change? That sounds scary.

Reply:
Regarding the first question, I'd say every component should ensure the consistency of their DOM structure. Should I add that to the RFC?
I personally see value in enforcing this because if we plan for our component library to be extensible in all the different ways we provide, keeping DOM structure stable helps with people who choose to extend it with that in mind.
What specifically about this sounds scary?